Basic/ClientDriven/Assets/Scripts/ServerPlayerMove.cs

```diff
@@ -31,42 +35,66 @@ public override void OnNetworkSpawn()
-        m_Client.SetSpawnClientRpc(spawnPoint.transform.position, new ClientRpcParams() { Send = new ClientRpcSendParams() { TargetClientIds = new []{OwnerClientId}}});
+        m_ClientPlayerMove.SetSpawnClientRpc(spawnPosition, 
+            new ClientRpcParams() { Send = new ClientRpcSendParams() { TargetClientIds = new []{OwnerClientId}}});
     }
 
-    private NetworkObject m_PickedUpObj;
-
     [ServerRpc]
     public void PickupObjServerRpc(ulong objToPickupID)
-    {
+    {        
         NetworkManager.SpawnManager.SpawnedObjects.TryGetValue(objToPickupID, out var objToPickup);
         if (objToPickup == null || objToPickup.transform.parent != null) return; // object already picked up, server authority says no
 
         objToPickup.GetComponent<Rigidbody>().isKinematic = true;
-        objToPickup.transform.parent = transform;
-        objToPickup.GetComponent<NetworkTransform>().InLocalSpace = true;
-        objToPickup.transform.localPosition = Vector3.up;
-        ObjPickedUp.Value = true;
-        m_PickedUpObj = objToPickup;
+        if (objToPickup.TryGetComponent(out NetworkObject networkObject) && networkObject.TrySetParent(transform))
+        {
+            objToPickup.GetComponent<NetworkTransform>().InLocalSpace = true;
+            objToPickup.transform.localPosition = m_LocalHeldPosition;
+            objToPickup.GetComponent<ServerIngredient>().ingredientDespawned += IngredientDespawned;
+            isObjectPickedUp.Value = true;
+            m_PickedUpObject = objToPickup;
+        }
     }
 
-    [ServerRpc]
-    public void DropObjServerRpc()
+    void IngredientDespawned()
+    {
+        m_PickedUpObject = null;
+        isObjectPickedUp.Value = false;
+    }
+
+    void FixedUpdate()
     {
-        if (m_PickedUpObj != null)
+        if (m_DropRequested)
         {
-            // can be null if enter drop zone while carying
-            m_PickedUpObj.transform.localPosition = new Vector3(0, 0, 2);
-            m_PickedUpObj.transform.parent = null;
-            m_PickedUpObj.GetComponent<Rigidbody>().isKinematic = false;
-            m_PickedUpObj.GetComponent<NetworkTransform>().InLocalSpace = false;
-            m_PickedUpObj = null;
+            m_DropRequested = false;
+
+            if (m_PickedUpObject != null)
+            {
+                // can be null if enter drop zone while carrying
+                m_PickedUpObject.transform.localPosition = m_LocalDropPosition;
+                m_PickedUpObject.transform.parent = null;
+                m_PickedUpObject.GetComponent<Rigidbody>().isKinematic = false;
+                m_PickedUpObject.GetComponent<NetworkTransform>().InLocalSpace = false;
+                m_PickedUpObject = null;
+            }
+            
+            isObjectPickedUp.Value = false;
         }
+    }
 
-        ObjPickedUp.Value = false;
+    [ServerRpc]
+    public void DropObjServerRpc()
+    {
+        m_DropRequested = true;
```

Comment:
When we were looking at this, I thought this didn't fix the issue, was that it in the end? Or did you forget to reverse that change?

Reply:
I guess because RPCs can happen in update, deferring physics changes to FixedUpdate makes sense still... nvm the above comment.
I'm still curious though whether that fixed the issue or not

Reply:
If it did, it'd be great to comment this and produce some edu comments.

Reply:
When we were trying things out, the changes we made were client side on sending RPCs to the server. This modification on the server side of code resolved the issue. 
Though what's curious to me is moving this code to a coroutine and waiting for the next FixedUpdate did not produce the same results. I want to follow up with SDK on this offline, but this will absolutely go into the updated doc, as a new section.

Reply:
before telling our users they should do this, it'd be great if we understood exactly why (and make sure this isn't an issue with the SDK that we're just building a workaround for --> which would need to be reverted a month or two later...)

Reply:
This has been since removed. The previous functionality of dropping an ingredient overhead was restored. This issue will be addressed (if need be) in the follow-up PR that integrates the pickup animations.